lib/zlib.js

```diff
@@ -594,7 +594,7 @@ Zlib.prototype._processChunk = function(chunk, flushFlag, cb) {
       this.callback = null;
     }
 
-    if (self._hadError)
+    if (self._hadError || !self._handle)
```

Comment:
I think `self._hadError` implies `self._handle`, so this might just be `if (!self._handle)`?

Reply:
If you want it that way, I can change it. However, I prefer it this way because:
1. This is a backport, so the impetus is on minimal changes rather than best code for future as the code is already obsolete
2. This makes it explicit (and easier to review) that the change can not cause additional breakage by removing some handling that needs to be explicitly checked that all sites which set `_hadError` also clear `_handle`.
3. In case `_hadError` behavior is changed due to some other case, relying on `_handle` might not be guaranteed

But, I don't really know the project well enough to know if my arguments are valid or not. So, either way is fine for me, just let me know if it would be better with just `if (!self._handle)`.